repo/initialize.go

```diff
@@ -163,6 +166,10 @@ func repositoryObjectFormatFromOptions(opt *NewRepositoryOptions) (*format.Repos
 		f.HMACSecret = nil
 	}
 
+	if opt.BlockFormat.ECCOverheadPercent == 0 {
+		f.ContentFormat.ECC = ""
+	}
+
```

Comment:
we should disallow ECC when `IndexVersion == 1`

Also, can you add a test or two using something like:

```
	e := testenv.NewCLITest(t, testenv.RepoFormatNotImportant, runner)

	defer e.RunAndExpectSuccess(t, "repo", "disconnect")
	e.RunAndExpectSuccess(t, "repo", "create", "filesystem", "--path", e.RepoDir, /* ECC parameters */)
	output := e.RunAndExpectSuccess(t, "repo", "status")
        // verify output contains ECC status line
```

I'm wondering if it makes sense to add few quick e2e tests for the ECC:

- create repo with 50% overhead, backup 1MB of random data, ensure the repo size is >= 1.5MB
- as a control, backup the same file with no ECC and assert that repository size is <=1.1MB.
- after backup with ECC, go through every single file in the repo (except `kopia.repository`) and flip one random bit in each file - the repo should still be readable

Reply:
Done